packages/@jsii/go-runtime/jsii-experimental/runtime.go

```diff
@@ -169,6 +185,38 @@ func StaticSet(fqn FQN, property string, value interface{}) {
 	}
 }
 
+func castValToRef(data interface{}) (objref, bool) {
+	ref := objref{}
+	ok := false
+	dataVal := reflect.ValueOf(data)
+
+	if dataVal.Kind() == reflect.Map {
+		for _, k := range dataVal.MapKeys() {
+			// Finding values type requires extracting from reflect.Value
+			// otherwise .Kind() returns `interface{}`
+			v := reflect.ValueOf(dataVal.MapIndex(k).Interface())
+
+			if k.Kind() == reflect.String && k.String() == "$jsii.byref" && v.Kind() == reflect.String {
+				ref.JsiiInstanceId = v.String()
+				ok = true
+			}
+
+		}
+	}
+
+	return ref, ok
+}
+
+// castSetToPtr accepts a pointer to any type and attempts to cast the value
+// argument to be the same type. Then it sets the value of the pointer element
+// to be the newly casted data. This is used to cast payloads from JSII to
+// expected return types for Get and Invoke functions.
+func castSetToPtr(ptr interface{}, data interface{}) {
```

Comment:
I was initially confused, wondering "what set are we talking about". And then I figured out... but I kinda still think we might be better off actually renaming this to: `castAndSetToPtr`.